### DISCUSSION TITLE
why doesn't `rg -F '-F'` work?

### DISCUSSION THREAD

**Discussion by AndyJado** [BurntSushi/ripgrep#2468] [Q&A] [answered]
version 13.0.0

```sh
rg -h | rg -F '-F'
```

**Comment by s-p-turner** [comment, marked as answer]
Would it work correctly if you added the -e/--regexp flag?

rg -h | rg -Fe '-F'

**Reply by BurntSushi** [reply]
Yes, this is correct. `rg -F '-F'` is no different than `rg -F -F`. ripgrep can't tell that the second `-F` is _intended_ to be your search pattern. This is mentioned right near the top of the man page:

```
            A regular expression used for searching. To match a pattern beginning with a
            dash, use the -e/--regexp flag.
            
            For example, to search for the literal '-foo', you can use this flag:
            
                rg -e -foo
            
            You can also use the special '--' delimiter to indicate that no more flags
            will be provided. Namely, the following is equivalent to the above:
            
                rg -- -foo
```

So, you can do `rg -Fe -F` as suggested above. Or `rg -F -e -F`. Or `rg -F -- -F`. Or to get a little more obscure, `rg -F '[-]F'` or even `rg -F '\x2DF'`.
